### PR TITLE
Sale data upgrader - Version 0.5

### PR DESCRIPTION
**Pull request by JordyMoos**
Version the sales data so we know when data is old and can upgrade the
data it to higher versions
- [x] Data versioning
- [x] Set current version
- [x] Add upgrade function
- [x] Test

### CODE REVIEW

File: JMGuildSaleHistoryTracker.lua [hunk 1]
```diff
@@ -250,6 +256,61 @@ function Indexer:addExistingDataToTheIndex()
...
+-- To check and update the sale data
+--
+function SaleUpgrader:upgrade()
+    if not SavedVariables.dataVersion then
+        SavedVariables.dataVersion = 0
+    end
+
+    while SavedVariables.dataVersion < Config.dataVersion do
+        local upgradeFunction = SaleUpgrader.upgradeFunctionVersion[SavedVariables.dataVersion]
+        upgradeFunction()
```
Comment (marcusklaas) [inline JMGuildSaleHistoryTracker.lua]: Should we not pass some data into this function, or does it act on global variables?

**Comment by marcusklaas** [comment]
Good approach by the way, this seems like a solid way to guarantee backwards compatibility.

File: JMGuildSaleHistoryTracker.lua (hunk 1 above)
Comment (JordyMoos) [inline JMGuildSaleHistoryTracker.lua]: I thought about that, to loop through the sales at forehand and then call the upgrade function for each sale. But that would limit me in what the function can do.

So i thought it is better to just call it and allow it to do whatever it wants.
The sales data in stored in SavedVariables which is global for this scope so the upgrade function can change that whatever he wants and not just the sales in it

Comment (JordyMoos) [inline JMGuildSaleHistoryTracker.lua]: I can also add a reference to the SavedVariables in the function call to make it more explicit that the meaning up the upgrade functions is changing the saved variables.

What do you think?

**Comment by JordyMoos** [comment]
Thanks!

I thought so to but was not sure if i oversaw something

File: JMGuildSaleHistoryTracker.lua (hunk 1 above)
Comment (marcusklaas) [inline JMGuildSaleHistoryTracker.lua]: Maybe the intention and effects of the function are a bit clearer that way, yes.

**Pull request merged**